pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -915,3 +930,49 @@ type NomadSDConfig struct {
 	// +optional
 	EnableHTTP2 *bool `json:"enableHTTP2,omitempty"`
 }
+type DockerSwarmSDConfig struct {
+	// Address of the Docker daemon
+	// +kubebuilder:validation:Pattern="^[a-zA-Z][a-zA-Z0-9+.-]*://.+$"
+	// +required
+	Host string `json:"host"`
+	// Role of the targets to retrieve. Must be `Services`, `Tasks`, or `Nodes`.
+	// +kubebuilder:validation:Enum=Services;Tasks;Nodes
+	// +required
+	Role string `json:"role"`
+	// The port to scrape metrics from, when `role` is nodes, and for discovered
+	// tasks and services that don't have published ports.
+	// +kubebuilder:validation:Minimum=0
```

Comment:
```suggestion
	// +kubebuilder:validation:Minimum=0
```	// +kubebuilder:validation:Maximum=65535
```